pwndbg/commands/spray.py

```diff
@@ -1,26 +1,57 @@
 import argparse
 
 import gdb
-from pwnlib.util.cyclic import cyclic_gen
+from pwnlib.util.cyclic import cyclic
 
+import pwndbg.color.message as M
 import pwndbg.commands
 
-parser = argparse.ArgumentParser(description="Spray memory with a cyclic_gen() generated values")
+parser = argparse.ArgumentParser(description="Spray memory with cyclic() generated values")
 parser.add_argument("addr", help="Address to spray")
-parser.add_argument("length", nargs="?", help="Length of byte sequence", type=int, default=0)
+parser.add_argument(
+    "length",
+    help="Length of byte sequence, when unspecified sprays until the end of vmmap which address belongs to",
+    type=int,
+    nargs="?",
+    default=0,
+)
+parser.add_argument("--value", help="Value to spray memory with", type=str, required=False)
 
 
 @pwndbg.commands.ArgparsedCommand(parser)
-def spray(addr, length) -> None:
+def spray(addr, length, value) -> None:
     if length == 0:
-        try:
-            last_addr = next(p for p in pwndbg.gdblib.vmmap.get() if addr in p).end
-        except StopIteration:
-            last_addr = addr + 1  # Throw exception later when trying to write
+        last_addr = addr
+        for p in pwndbg.gdblib.vmmap.get():
+            if addr in p:
+                last_addr = p.end
+                break
         length = last_addr - int(addr)
 
-    valueBytes = cyclic_gen().get(length)
+        if length == 0:
+            print(M.error("Invalid address, can't find end of vmmap"))
+            return
+
+    value_bytes = b""
+
+    if value:
+        if value.startswith("0x"):
+            strlen = len(value[2:])
+            if strlen & 1:  # 0x1, 0x123...
+                strlen += 1
+            value_bytes = int(value, 16).to_bytes(int(strlen / 2), byteorder="big")
```

Comment:
See if this works instead:
```suggestion
            value_bytes = int(value, 16).to_bytes((len(value[2:]) + 1) // 2, byteorder="big")
```